src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -295,47 +310,106 @@ private static MethodInfo GetMethodInfo<T>(Expression<T> expr)
         {
             FindApplicableConstructor(instanceType, argumentTypes, out ConstructorInfo constructor, out int?[] parameterMap);
 
+#if NET8_0_OR_GREATER
+            ConstructorInvoker invoker = ConstructorInvoker.Create(constructor);
+#endif
+
             ParameterInfo[] constructorParameters = constructor.GetParameters();
             if (constructorParameters.Length == 0)
             {
                 return (IServiceProvider serviceProvider, object?[]? arguments) =>
+#if NET8_0_OR_GREATER
+                    invoker.Invoke()!;
```

Comment:
```suggestion
                    invoker.Invoke();
```

ConstructorInvoker.Invoke() returns `object`. No need to suppress nullable here.

Reply:
At the time this PR was created, there was a nullability issue with ConstructorInfo.

Reply:
Sync'd to main and removed those.